### PR TITLE
Compatibility with python 3.12

### PR DESCRIPTION
**Pull request by hoanphungt**

### CODE REVIEW

**Review by benvanbasten-ns — APPROVED** [review]
Don't forget to update README

**Comment by hoanphungt** [comment]
@leendertvanwolfswinkel 
Hi Leendert, I upgraded some wheels for the 3Di Result Analysis to make the plugin compatible with Python 3.12 (used by both QGIS LTR and non LTR). Could you include it into your release plan please? It's needed for the Rana Desktop Client.

**Comment by leendertvanwolfswinkel** [comment]
I tested it, but I get an error as soon as I activate the plugin

```
ImportError: cannot import name '_errors' from partially initialized module 'h5py' (most likely due to a circular import) (C:\Users\leendert.vanwolfswin\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\threedi_results_analysis\deps\h5py\__init__.py) 
Traceback (most recent call last):
  File "C:\PROGRA~1/QGIS33~2.3/apps/qgis/./python\qgis\utils.py", line 404, in loadPlugin
    __import__(packageName)
  File "C:\PROGRA~1/QGIS33~2.3/apps/qgis/./python\qgis\utils.py", line 923, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users/leendert.vanwolfswin/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\threedi_results_analysis\__init__.py", line 58, in 
    dependencies.check_importability()
  File "C:\Users/leendert.vanwolfswin/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\threedi_results_analysis\dependencies.py", line 386, in check_importability
    imported_package = importlib.import_module(package)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\PROGRA~1\QGIS33~2.3\apps\Python312\Lib\importlib\__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\leendert.vanwolfswin\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\threedi_results_analysis\deps\h5netcdf\__init__.py", line 16, in 
    from .core import CompatibilityError, Dimension, File, Group, Variable  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\PROGRA~1/QGIS33~2.3/apps/qgis/./python\qgis\utils.py", line 923, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\leendert.vanwolfswin\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\threedi_results_analysis\deps\h5netcdf\core.py", line 9, in 
    import h5py
  File "C:\PROGRA~1/QGIS33~2.3/apps/qgis/./python\qgis\utils.py", line 923, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\leendert.vanwolfswin\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\threedi_results_analysis\deps\h5py\__init__.py", line 25, in 
    from . import _errors
ImportError: cannot import name '_errors' from partially initialized module 'h5py' (most likely due to a circular import) (C:\Users\leendert.vanwolfswin\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\threedi_results_analysis\deps\h5py\__init__.py)
```

**Comment by hoanphungt** [comment]
@leendertvanwolfswinkel 
It probably because of the version of `h5py` and `scipy` that are not supported for python 3.12 yet. For me, I didn't have this issue since it automatically downloads a version that has support for python 3.12. I made a new commit to update these 2 packages. Could you please try to test again?

File: dependencies.py [hunk 1]
```diff
@@ -87,12 +87,12 @@
     H5PY_DEPENDENCY = Dependency("h5py", "h5py", "==2.10.0", False)
 else:
     SUPPORTED_HDF5_VERSIONS = ["1.14.0"]
-    H5PY_DEPENDENCY = Dependency("h5py", "h5py", "==3.8.0", True)
+    H5PY_DEPENDENCY = Dependency("h5py", "h5py", "==3.10.0", True)
 
 WINDOWS_PLATFORM_DEPENDENCIES = [Dependency("scipy", "scipy", "==1.6.2", False)]
 if QGIS_VERSION >= 32811 and platform.system() == "Windows":
```
Comment (benvanbasten-ns) [inline dependencies.py]: Perhaps you should use this version check. In that case the plugin will also work in older versions?

File: external-dependencies/populate.sh [hunk 2]
```diff
@@ -40,16 +40,16 @@ mkdir build
 cd build
 
 # Download the custom compiled qgis version tar of h5py, create a tar from the distro subfolder
-wget http://download.osgeo.org/osgeo4w/v2/x86_64/release/python3/python3-h5py/python3-h5py-3.8.0-1.tar.bz2
-tar -xvf python3-h5py-3.8.0-1.tar.bz2
-tar -cf h5py-3.8.0.tar -C ./apps/Python39/Lib/site-packages/ .
-cp h5py-3.8.0.tar ..
+wget http://download.osgeo.org/osgeo4w/v2/x86_64/release/python3/python3-h5py/python3-h5py-3.10.0-1.tar.bz2
```
Comment (benvanbasten-ns) [inline external-dependencies/populate.sh:50]: Its a choice, but perhaps download both versions so it works in older versions?

**Review by benvanbasten-ns — APPROVED** [review]
For a prototype it seems fine, but it needs to be tested on windows, I guess.

**Pull request merged**